src/Components/pages/Host.jsx

```diff
@@ -0,0 +1,245 @@
+
+function fetchGame(gameId, callback) {
+    return fire.database().ref('/Rooms').orderByChild('gameId').equalTo(gameId).once('value',callback);
+}
+
+class Host extends Component {
+    /*
+    Possible phases:
+    - not-joined : display joining game
+    - connection : players joining phase
+    - question : displaying questions phase
+    - between-question : display a page between questions
+    - leaderboards : display the winners at the end
+    - ended : the game has ended
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            phase: 'not-joined',
+            questionNum: 0,
+            gameId: null,
+            password: '',
+            authenticated: 'no',
+            listening: 'no',
+            users: [],
+        }
+        this.joinGame = this.joinGame.bind(this);
+        this.updatePhase = this.updatePhase.bind(this);
+        this.initGameListener = this.initGameListener.bind(this);
+        this.restartGame = this.restartGame.bind(this);
+        this.quitGame = this.quitGame.bind(this);
+        this.endGame = this.endGame.bind(this);
+    }
+
+    componentDidMount() {
+        const {gameId} = this.state;
+    }
+
+    componentDidChange() {
+    }
+
+    handleChangeSelect = (event) => {
+        this.setState({[event.target.name]:event.target.value});
+    };
+
+    handleChange = name => (event) => {
+        this.setState({
+            [name]:event.target.value,
+        });
+    }
+
+    updatePhase(gameupdate) {
+        const {gameId} = this.state;
+        this.setState({
+            phase: gameupdate,
+        })
+        db.collection('Rooms').doc(gameId).update({
+            phase: gameupdate,
+        });
+    }
+
+    restartGame() {
+
+    }
+
+    quitGame() {
+        const {toggleHeader} = this.props;
+        this.updateGame({phase:null});
+    }
+
+    endGame(){
+        this.updateGame({phase:'final_result'});
+    }
+
+    addDummyUser(nickname) {
+        const {gameId} = this.state;
+        const gameRef = db.collection('Rooms').doc(gameId);
+        const userRef = gameRef.collection('users').doc(nickname);
+
+        const gameInfo = {
+            nickname: nickname,
+            investments: [],
+            personal_value: -1,
+            money_left: -1,
+            gains: 0,
+            losses: 0,
+        }
+
+        userRef.set(gameInfo);
+    }
+
+    initGameListener() {
+        var gameRef;
+        gameRef = db.collection('Rooms').doc(this.state.gameId);
+        const that = this;
+
+        gameRef.on('value',(snapshot) => {
+            const game = snapshot.val();
+            if (!game.phase) {
+                game.phase = 'setup';
+            }
+            if (game) {
+                that.setState({
+                    game,
+                });
+            } else {
+                that.setState({
+                    game: '',
+                });
+            }
+        });
+    }
+
+    joinGame() {
```

Comment:
can @jack-y-xu add a comment at the top of this method to describe exactly what is going on?